core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -861,7 +861,14 @@ case class DefBlackBox(
   params: Map[String, Param])
     extends Component
 
-case class Circuit(name: String, components: Seq[Component], annotations: Seq[ChiselAnnotation], renames: RenameMap) {
-  def firrtlAnnotations: Iterable[Annotation] = annotations.flatMap(_.toFirrtl.update(renames))
-
+case class Circuit(
+  name:           String,
+  components:     Seq[Component],
+  annotations:    Seq[ChiselAnnotation],
+  renames:        RenameMap,
+  newAnnotations: Seq[ChiselToFirrtlAnnotations] = Seq.empty) {
+  def firrtlAnnotations: Iterable[Annotation] =
```

Comment:
> Shouldn't there be something when we do Convert to FIRRTL that actually looks at these newAnnotations?

@mwachs5  - that's handled here - in short firrtlAnnotations now includes flattening of the new Seq of annotations in addition to the ones from the legacy API.

Reply:
This is why I originally added this def, because the annotations need to be renamed anyway (for DataViews) so it's pretty important people don't access the ChiselAnnotations from the class directly.

In fact, could you deprecate the `annotations` field of this class to tell anyone who happens to be using this API to use `firrtlAnnotations` instead?

Reply:
Done - added the deprecated warning on both `annotations` and `newAnnotations` just in case someone tries to use that as well. Hope that's fine.